### ISSUE TITLE
Old low hanging fruit for user facing issue: Awstats and Webalizer assets in public html and why it should be evaluated.,

### ISSUE THREAD

**Issue by abclution** [virtualmin/virtualmin-gpl#220]
Having the statistics programs creating files in the document root is a multifaceted issue. Let me quickly run through my thoughts.

From the point of an end user, oftentimes when starting to run my own website, I have very little clue to how everything interacts. Being given an hosting account, and told to put my web files into my public_html folder, upon visiting that folder, with statistics enabled there is a set of *strange* files there. 

The list of files there varies but it is usually (from the perspective of an end user...)

- public_html/stats/ - a folder called stats (Webalizer)

- public_html/icon/ - a folder called icon with random images in it (Remember I am a brand new end user)

- public_html/awstats-icon - a folder that has the same information as the icon folder

- public_html/awstatsicons - again a folder that has the same images as the icon folder

Oftentimes, a user will just see these presumably useless and confusing folders and delete them. (50 percent of the users will delete them, and the other half are too scared to touch them.) 

This is fine for the Awstats assets. Awstats keeps its generated reports in ~/awstats

**But for the Webalyzer public_html/stats folder, this is somewhat catastrophic as all the reports that it has generated historically is stored there and unless log rotation is off, there is no way to regenerate this reports (Correct me if I am wrong!)**

To clear up the document root, create less confusion for new users, and generally provide a cleaner and clearer starting point for new users, while additionally protecting the historical Webalyzer statistics somewhat from hamfisted users...

I propose moving the stats folder into the domain home ~/stats_webalizer/ and using apache alias + directory stanza / directive for providing the access from their http://domain.tld/stats

I have done this in my own custom server template and it works well.

example below

```
<Directory  ${HOME}/stats_webalizer>
Options -Indexes +IncludesNOEXEC +SymLinksifOwnerMatch
allow from all
Require all granted
</Directory>
Alias /stats  ${HOME}/stats_webalizer
```

Additionally while we are at it, lets clean up the awstats assets as well from the document root as well.

```
<Directory  /usr/share/awstats/icon>
Options -Indexes +IncludesNOEXEC +SymLinksifOwnerMatch
allow from all
Require all granted
</Directory>
Alias /awstats-icon /usr/share/awstats/icon
Alias /awstatsicons /usr/share/awstats/icon
```

No additional changes need to be done as far as I can tell. It would be prudent to not create the various stats assets, stats,icon, awstats-icon, awstatsicons folders any longer in the domain creation time, but even if they are created and the user deletes them, their historical webalizer statistics will be safe and awstats will still be functioning.

Feedback? Thoughts?

**Comment by jcameron** [comment]
This is a good suggestion, at least partially. We can certainly move the AWstats and Webalizer data files outside of `public_html` ... however, other files like those `icons` directories cannot be moved, as they contain images used in the AWstats web UI. An Apache alias like you suggested might work, but we'd need to be careful to ensure that permissions are set correctly so that the common directory is accessible by all domains.

**Comment by abclution** [comment]
Default permissions on my system were correct. (rx for other)

**Comment by jcameron** [comment]
I'll take a look into this, and update this ticket with progress..

**Comment by abclution** [comment]
Nifty! I think it will improve end user experience (as well as omg I accidentally the whole statistics situations... )

**Comment by jcameron** [comment]
FYI, I just realized that Virtualmin already supports moving the Webalizer `stats` dir to another location, via the setting at System Settings -> Server Templates -> Default Settings -> Website for domain -> Subdirectory for Webalizer statistics.

**Comment by chris001** [comment]
> FYI, I just realized that Virtualmin already supports moving the Webalizer `stats` dir to another location, via the setting at System Settings -> Server Templates -> Default Settings -> Website for domain -> Subdirectory for Webalizer statistics.

Yes, as @abclution mentioned in the original issue, Webmin has a setting for "Subdirectory for Webalizer statistics".  The issue is, Webmin/Virtualmin is using a bad default value in the server template (`www/stats`) for this value, instead it should set a better default value, to "Directory under home (`stats_webalizer`)" + the Apache server templating stuff for the redirects from various directory names to the one directory, etc.  See original issue!
